Can't Build

I got the following error when I tried to build the source.
```
build.zig:23:40: error: expected token 'Symbol', found '{'
    const common_params = [][]const u8.{
                                       ^
```

Anyone know what the problem could be?
Running Windows 10 Pro 64 bit, i3 7100T

--- a/build.zig
+++ b/build.zig
@@ -20,13 +20,13 @@ pub fn build(b: *Builder) void {
     const qemu       = b.step("qemu",       "Run the OS with Qemu");
     const qemu_debug = b.step("qemu-debug", "Run the OS with Qemu and wait for debugger to attach");
 
-    const common_params = [][]const u8.{
+    const common_params = [][]const u8 {
         "qemu-system-i386",
         "-display", "curses",
         "-kernel", kernel,
         "-initrd", join(b.allocator, ',', terminal, keyboard, shell) catch unreachable,
     };
-    const debug_params = [][]const u8.{"-s", "-S"};
+    const debug_params = [][]const u8 {"-s", "-S"};
 
     var qemu_params       = Array([]const u8).init(b.allocator);
     var qemu_debug_params = Array([]const u8).init(b.allocator);

--- a/kernel/elf.zig
+++ b/kernel/elf.zig
@@ -2,7 +2,7 @@ const vmem = @import("vmem.zig");
 const x86 = @import("x86.zig");
 
 // Description of an ELF executable.
-const ELFHeader = packed struct.{
+const ELFHeader = packed struct {
     e_ident:     [16]u8,
     e_type:      u16,
     e_machine:   u16,
@@ -35,7 +35,7 @@ const PF_W = 0x2;
 const PF_R = 0x4;
 
 // Description of an ELF program segment.
-const ELFProgHeader = packed struct.{
+const ELFProgHeader = packed struct {
     p_type:   u32,
     p_offset: u32,
     p_vaddr:  u32,

--- a/kernel/gdt.zig
+++ b/kernel/gdt.zig
@@ -24,7 +24,7 @@ const PROTECTED = (1 << 2);
 const BLOCKS_4K = (1 << 3);
 
 // Structure representing an entry in the GDT.
-const GDTEntry = packed struct.{
+const GDTEntry = packed struct {
     limit_low:  u16,
     base_low:   u16,
     base_mid:   u8,
@@ -35,13 +35,13 @@ const GDTEntry = packed struct.{
 };
 
 // GDT descriptor register.
-const GDTRegister = packed struct.{
+const GDTRegister = packed struct {
     limit: u16,
     base: *const GDTEntry,
 };
 
 // Task State Segment.
-const TSS = packed struct.{
+const TSS = packed struct {
     unused1: u32,
     esp0:    u32,      // Stack to use when coming to ring 0 from ring > 0.
     ss0:     u32,      // Segment to use when coming to ring 0 from ring > 0.
@@ -60,7 +60,7 @@ const TSS = packed struct.{
 //     flags: Segment flags.
 //
 fn makeEntry(base: usize, limit: usize, access: u8, flags: u4) GDTEntry {
-    return GDTEntry.{ .limit_low  = @truncate(u16,  limit       ),
+    return GDTEntry { .limit_low  = @truncate(u16,  limit       ),
                       .base_low   = @truncate(u16,  base        ),
                       .base_mid   = @truncate(u8,   base   >> 16),
                       .access     = @truncate(u8,   access      ),
@@ -70,7 +70,7 @@ fn makeEntry(base: usize, limit: usize, access: u8, flags: u4) GDTEntry {
 }
 
 // Fill in the GDT.
-var gdt align(4) = []GDTEntry.{
+var gdt align(4) = []GDTEntry {
     makeEntry(0, 0, 0, 0),
     makeEntry(0, 0xFFFFF, KERNEL | CODE, PROTECTED | BLOCKS_4K),
     makeEntry(0, 0xFFFFF, KERNEL | DATA, PROTECTED | BLOCKS_4K),
@@ -80,17 +80,17 @@ var gdt align(4) = []GDTEntry.{
 };
 
 // GDT descriptor register pointing at the GDT.
-var gdtr = GDTRegister.{
+var gdtr = GDTRegister {
     .limit = u16(@sizeOf(@typeOf(gdt))),
     .base  = &gdt[0],
 };
 
 // Instance of the Task State Segment.
-var tss = TSS.{
+var tss = TSS {
     .unused1 = 0,
     .esp0    = undefined,
     .ss0     = KERNEL_DATA,
-    .unused2 = []u32.{ 0 } ** 22,
+    .unused2 = []u32 { 0 } ** 22,
     .unused3 = 0,
     .iomap_base = @sizeOf(TSS),
 };

--- a/kernel/idt.zig
+++ b/kernel/idt.zig
@@ -8,7 +8,7 @@ pub const INTERRUPT_GATE = 0x8E;
 pub const SYSCALL_GATE   = 0xEE;
 
 // Structure representing an entry in the IDT.
-const IDTEntry = packed struct.{
+const IDTEntry = packed struct {
     offset_low:  u16,
     selector:    u16,
     zero:        u8,
@@ -17,7 +17,7 @@ const IDTEntry = packed struct.{
 };
 
 // IDT descriptor register.
-const IDTRegister = packed struct.{
+const IDTRegister = packed struct {
     limit: u16,
     base:  *[256]IDTEntry,
 };
@@ -26,7 +26,7 @@ const IDTRegister = packed struct.{
 var idt: [256]IDTEntry = undefined;
 
 // IDT descriptor register pointing at the IDT.
-const idtr = IDTRegister.{
+const idtr = IDTRegister {
     .limit = u16(@sizeOf(@typeOf(idt))),
     .base  = &idt,
 };

--- a/kernel/interrupt.zig
+++ b/kernel/interrupt.zig
@@ -1,3 +1,4 @@
+// zig fmt: off
 const std = @import("std");
 const isr = @import("isr.zig");
 const scheduler = @import("scheduler.zig");

--- a/kernel/interrupt.zig
+++ b/kernel/interrupt.zig
@@ -31,9 +31,9 @@ const IRQ_15 = IRQ_0 + 15;
 const SYSCALL = 128;
 
 // Registered interrupt handlers.
-var handlers = []fn()void.{ unhandled } ** 48;
+var handlers = []fn()void { unhandled } ** 48;
 // Registered IRQ subscribers.
-var irq_subscribers = []MailboxId.{ MailboxId.Kernel } ** 16;
+var irq_subscribers = []MailboxId { MailboxId.Kernel } ** 16;
 
 ////
 // Default interrupt handler.

--- a/kernel/ipc.zig
+++ b/kernel/ipc.zig
@@ -15,7 +15,7 @@ const Thread = thread.Thread;
 const ThreadQueue = thread.ThreadQueue;
 
 // Structure representing a mailbox.
-pub const Mailbox = struct.{
+pub const Mailbox = struct {
     messages:      List(Message),
     waiting_queue: IntrusiveList(Thread, "queue_link"),
     // TODO: simplify once #679 is resolved.
@@ -27,7 +27,7 @@ pub const Mailbox = struct.{
     //     An empty mailbox.
     //
     pub fn init() Mailbox {
-        return Mailbox.{
+        return Mailbox {
             .messages = List(Message).init(),
             .waiting_queue = ThreadQueue.init(),
         };
@@ -146,7 +146,7 @@ fn processOutgoingMessage(message: *const Message) Message {
     var message_copy = message.*;
 
     switch (message.sender) {
-        MailboxId.This => message_copy.sender = MailboxId.{ .Thread = (scheduler.current().?).tid },
+        MailboxId.This => message_copy.sender = MailboxId { .Thread = (scheduler.current().?).tid },
         // MailboxId.Port   => TODO: ensure the sender owns the port.
         // MailboxId.Kernel => TODO: ensure the sender is really the kernel.
         else => {},

--- a/kernel/ipc.zig
+++ b/kernel/ipc.zig
@@ -1,3 +1,4 @@
+// zig fmt: off
 const std = @import("std");
 const layout = @import("layout.zig");
 const mem = @import("mem.zig");
@@ -7,7 +8,7 @@ const scheduler = @import("scheduler.zig");
 const thread = @import("thread.zig");
 const x86 = @import("x86.zig");
 const HashMap = std.HashMap;
-const IntrusiveList = std.IntrusiveLinkedList;
+const LinkedList = std.LinkedList;
 const List = std.LinkedList;
 const MailboxId = std.os.zen.MailboxId;
 const Message = std.os.zen.Message;
@@ -17,7 +18,7 @@ const ThreadQueue = thread.ThreadQueue;
 // Structure representing a mailbox.
 pub const Mailbox = struct {
     messages:      List(Message),
-    waiting_queue: IntrusiveList(Thread, "queue_link"),
+    waiting_queue: LinkedList(void),
     // TODO: simplify once #679 is resolved.
 
     ////
@@ -71,12 +72,12 @@ pub fn getOrCreatePort(id: u16) *Mailbox {
 pub fn send(message: *const Message) void {
     // NOTE: We need a copy in kernel space, because we
     // are potentially switching address spaces.
-    const message_copy = processOutgoingMessage(message);  // FIXME: should this be volatile?
+    const message_copy = processOutgoingMessage(message.*);  // FIXME: should this be volatile?
     const mailbox = getMailbox(message.receiver);
 
     if (mailbox.waiting_queue.popFirst()) |first| {
         // There's a thread waiting to receive, wake it up.
-        const receiving_thread = first.toData();
+        const receiving_thread = @fieldParentPtr(Thread, "queue_link", first);
         scheduler.new(receiving_thread);
         // Deliver the message into the receiver's address space.
         deliverMessage(message_copy);
@@ -122,8 +123,8 @@ pub fn receive(destination: *Message) void {
 // Returns:
 //     The address of the mailbox.
 //
-fn getMailbox(mailbox_id: *const MailboxId) *Mailbox {
-    return switch (mailbox_id.*) {
+fn getMailbox(mailbox_id: MailboxId) *Mailbox {
+    return switch (mailbox_id) {
         MailboxId.This   => &(scheduler.current().?).mailbox,
         MailboxId.Thread => |tid| &(thread.get(tid).?).mailbox,
         MailboxId.Port   => |id| getOrCreatePort(id),
@@ -142,8 +143,8 @@ fn getMailbox(mailbox_id: *const MailboxId) *Mailbox {
 // Returns:
 //     A copy of the message, post processing.
 //
-fn processOutgoingMessage(message: *const Message) Message {
-    var message_copy = message.*;
+fn processOutgoingMessage(message: Message) Message {
+    var message_copy = message;
 
     switch (message.sender) {
         MailboxId.This => message_copy.sender = MailboxId { .Thread = (scheduler.current().?).tid },
@@ -178,12 +179,12 @@ fn processOutgoingMessage(message: *const Message) Message {
 // Arguments:
 //     message: The message to be delivered.
 //
-fn deliverMessage(message: *const Message) void {
+fn deliverMessage(message: Message) void {
     const receiver_thread = scheduler.current().?;
     const destination = receiver_thread.message_destination;
 
     // Copy the message structure.
-    destination.* = message.*;
+    destination.* = message;
 
     // Map the message's payload into the thread's address space.
     if (message.payload) |payload| {

--- a/kernel/isr.zig
+++ b/kernel/isr.zig
@@ -16,7 +16,7 @@ extern fn isr44()void; extern fn isr45()void; extern fn isr46()void; extern fn i
 extern fn isr128()void;
 
 // Context saved by Interrupt Service Routines.
-pub const Context = packed struct.{
+pub const Context = packed struct {
     registers: Registers,  // General purpose registers.
 
     interrupt_n: u32,  // Number of the interrupt.
@@ -36,12 +36,12 @@ pub const Context = packed struct.{
 };
 
 // Structure holding general purpose registers as saved by PUSHA.
-pub const Registers = packed struct.{
+pub const Registers = packed struct {
     edi: u32, esi: u32, ebp: u32, esp: u32,
     ebx: u32, edx: u32, ecx: u32, eax: u32,
 
     pub fn init() Registers {
-        return Registers.{
+        return Registers {
             .edi = 0, .esi = 0, .ebp = 0, .esp = 0,
             .ebx = 0, .edx = 0, .ecx = 0, .eax = 0,
         };

--- a/kernel/mem.zig
+++ b/kernel/mem.zig
@@ -6,7 +6,7 @@ const assert = @import("std").debug.assert;
 const Color = tty.Color;
 
 // Standard allocator interface.
-pub var allocator = mem.Allocator.{
+pub var allocator = mem.Allocator {
     .allocFn   = alloc,
     .reallocFn = realloc,
     .freeFn    = free,
@@ -16,7 +16,7 @@ var heap: []u8 = undefined;          // Global kernel heap.
 var free_list: ?*Block = undefined;  // List of free blocks in the heap.
 
 // Structure representing a block in the heap.
-const Block = struct.{
+const Block = struct {
     free: bool,     // Is the block free?
 
     prev: ?*Block,  // Adjacent block to the left.
@@ -33,7 +33,7 @@ const Block = struct.{
     //     The biggest possible free block.
     //
     pub fn init() Block {
-        return Block.{
+        return Block {
             .free = true,
             .prev = null,
             .next = null,
@@ -203,7 +203,7 @@ fn splitBlock(block: *Block, left_sz: usize) void {
 
     // Setup the second block at the end of the first one.
     var right_block = @intToPtr(*Block, @ptrToInt(block) + @sizeOf(Block) + left_sz);
-    right_block.* = Block.{
+    right_block.* = Block {
         .free = false,  // For consistency: not free until added to the free list.
         .prev = block,
         .next = block.next,

--- a/kernel/multiboot.zig
+++ b/kernel/multiboot.zig
@@ -1,3 +1,4 @@
+// zig fmt: off
 const tty = @import("tty.zig");
 const cstr = @import("std").cstr;
 const Process = @import("process.zig").Process;
@@ -113,7 +114,7 @@ const MultibootHeader = packed struct {
 // NOTE: this structure is incomplete.
 
 // Place the header at the very beginning of the binary.
-export const multiboot_header align(4) section(".multiboot") = multiboot: {
+export const multiboot_header align(4) linksection(".multiboot") = multiboot: {
     const MAGIC   = u32(0x1BADB002);  // Magic number for validation.
     const ALIGN   = u32(1 << 0);      // Align loaded modules.
     const MEMINFO = u32(1 << 1);      // Receive a memory map from the bootloader.

--- a/kernel/multiboot.zig
+++ b/kernel/multiboot.zig
@@ -11,7 +11,7 @@ pub const MULTIBOOT_INFO_MEMORY      = 0x00000001;
 pub const MULTIBOOT_INFO_MEM_MAP     = 0x00000040;
 
 // System information structure passed by the bootloader.
-pub const MultibootInfo = packed struct.{
+pub const MultibootInfo = packed struct {
     // Multiboot info version number.
     flags: u32,
 
@@ -88,14 +88,14 @@ pub const MULTIBOOT_MEMORY_AVAILABLE = 1;
 pub const MULTIBOOT_MEMORY_RESERVED  = 2;
 
 // Entries in the memory map.
-pub const MultibootMMapEntry = packed struct.{
+pub const MultibootMMapEntry = packed struct {
     size: u32,
     addr: u64,
     len:  u64,
     type: u32,
 };
 
-pub const MultibootModule = packed struct.{
+pub const MultibootModule = packed struct {
     // The memory used goes from bytes 'mod_start' to 'mod_end-1' inclusive.
     mod_start: u32,
     mod_end:   u32,
@@ -105,7 +105,7 @@ pub const MultibootModule = packed struct.{
 };
 
 // Multiboot structure to be read by the bootloader.
-const MultibootHeader = packed struct.{
+const MultibootHeader = packed struct {
     magic:    u32,  // Must be equal to header magic number.
     flags:    u32,  // Feature flags.
     checksum: u32,  // Above fields plus this one must equal 0 mod 2^32.
@@ -119,7 +119,7 @@ export const multiboot_header align(4) section(".multiboot") = multiboot: {
     const MEMINFO = u32(1 << 1);      // Receive a memory map from the bootloader.
     const FLAGS   = ALIGN | MEMINFO;  // Combine the flags.
 
-    break :multiboot MultibootHeader.{
+    break :multiboot MultibootHeader {
         .magic    = MAGIC,
         .flags    = FLAGS,
         .checksum = ~(MAGIC +% FLAGS) +% 1,

--- a/kernel/process.zig
+++ b/kernel/process.zig
@@ -13,11 +13,11 @@ var next_pid: u16 = 1;
 
 // Structure representing a process.
 pub const Process = struct {
-    pid:            u16,
+    pid: u16,
     page_directory: usize,
 
     next_local_tid: u8,
-    threads:        ThreadList,
+    threads: ThreadList,
 
     ////
     // Create a new process and switch to it.
@@ -30,11 +30,11 @@ pub const Process = struct {
     //
     pub fn create(elf_addr: usize, args: ?[]const []const u8) *Process {
         var process = mem.allocator.createOne(Process) catch unreachable;
-        process.* = Process {
-            .pid            = next_pid,
+        process.* = Process{
+            .pid = next_pid,
             .page_directory = vmem.createAddressSpace(),
             .next_local_tid = 1,
-            .threads        = ThreadList.init(),
+            .threads = ThreadList.init(),
         };
         next_pid += 1;
 
@@ -44,7 +44,7 @@ pub const Process = struct {
         const entry_point = elf.load(elf_addr);
         // ...and start executing it.
         const main_thread = process.createThread(entry_point);
-        insertArguments(main_thread, args orelse [][]const u8 {});
+        insertArguments(main_thread, args orelse [][]const u8{});
 
         return process;
     }
@@ -53,7 +53,7 @@ pub const Process = struct {
     // Destroy the process.
     //
     pub fn destroy(self: *Process) void {
-        assert (scheduler.current_process == self);
+        assert(scheduler.current_process == self);
 
         // Deallocate all of user space.
         vmem.destroyAddressSpace();
@@ -65,7 +65,7 @@ pub const Process = struct {
             // before deallocating the current one.
             it = node.next;
 
-            const thread = node.toData();
+            const thread = @fieldParentPtr(Thread, "process_link", node);
             thread.destroy();
         }
 

--- a/kernel/process.zig
+++ b/kernel/process.zig
@@ -12,7 +12,7 @@ const ThreadList = @import("thread.zig").ThreadList;
 var next_pid: u16 = 1;
 
 // Structure representing a process.
-pub const Process = struct.{
+pub const Process = struct {
     pid:            u16,
     page_directory: usize,
 
@@ -30,7 +30,7 @@ pub const Process = struct.{
     //
     pub fn create(elf_addr: usize, args: ?[]const []const u8) *Process {
         var process = mem.allocator.createOne(Process) catch unreachable;
-        process.* = Process.{
+        process.* = Process {
             .pid            = next_pid,
             .page_directory = vmem.createAddressSpace(),
             .next_local_tid = 1,
@@ -44,7 +44,7 @@ pub const Process = struct.{
         const entry_point = elf.load(elf_addr);
         // ...and start executing it.
         const main_thread = process.createThread(entry_point);
-        insertArguments(main_thread, args orelse [][]const u8.{});
+        insertArguments(main_thread, args orelse [][]const u8 {});
 
         return process;
     }

--- a/kernel/scheduler.zig
+++ b/kernel/scheduler.zig
@@ -1,3 +1,4 @@
+// zig fmt: off
 const gdt = @import("gdt.zig");
 const isr = @import("isr.zig");
 const mem = @import("mem.zig");
@@ -18,7 +19,7 @@ var ready_queue: ThreadQueue = undefined;       // Queue of threads ready for ex
 fn schedule() void {
     if (ready_queue.popFirst()) |next| {
         ready_queue.append(next);
-        const next_thread = next.toData();
+        const next_thread = @fieldParentPtr(Thread, "queue_link", next);
 
         contextSwitch(next_thread);
     }
@@ -89,7 +90,7 @@ pub fn enqueue(thread: *Thread) void {
 pub fn dequeue() ?*Thread {
     const thread = ready_queue.pop() orelse return null;
     schedule();
-    return thread.toData();
+    return @fieldParentPtr(Thread, "queue_link", thread);
 }
 
 ////
@@ -111,7 +112,7 @@ pub fn remove(thread: *Thread) void {
 //
 pub inline fn current() ?*Thread {
     const last = ready_queue.last orelse return null;
-    return last.toData();
+    return @fieldParentPtr(Thread, "queue_link", last);
 }
 
 ////

--- a/kernel/syscall.zig
+++ b/kernel/syscall.zig
@@ -10,7 +10,7 @@ const x86 = @import("x86.zig");
 const TypeId = @import("builtin").TypeId;
 
 // Registered syscall handlers.
-pub var handlers = []fn()void.{
+pub var handlers = []fn()void {
     SYSCALL(exit),                    // 0
     SYSCALL(ipc.send),                // 1
     SYSCALL(ipc.receive),             // 2
@@ -35,7 +35,7 @@ pub var handlers = []fn()void.{
 fn SYSCALL(comptime function: var) fn()void {
     const signature = @typeOf(function);
 
-    return struct.{
+    return struct {
         // Return the n-th argument passed to the function.
         fn arg(comptime n: u8) @ArgType(signature, n) {
             return getArg(n, @ArgType(signature, n));

--- a/kernel/thread.zig
+++ b/kernel/thread.zig
@@ -24,7 +24,7 @@ pub const ThreadList  = List(Thread, "process_link");
 pub const ThreadQueue = List(Thread, "queue_link");
 
 // Structure representing a thread.
-pub const Thread = struct.{
+pub const Thread = struct {
     // TODO: simplify once #679 is solved.
     process_link: List(Thread, "process_link").Node,
     queue_link:   List(Thread, "queue_link").Node,
@@ -57,7 +57,7 @@ pub const Thread = struct.{
 
         // Allocate and initialize the thread structure.
         const thread = mem.allocator.createOne(Thread) catch unreachable;
-        thread.* = Thread.{
+        thread.* = Thread {
             .context      = initContext(entry_point, stack),
             .process      = process,
             .local_tid    = local_tid,
@@ -120,7 +120,7 @@ fn initContext(entry_point: usize, stack: usize) isr.Context {
     var stack_top = @intToPtr(*usize, stack + STACK_SIZE - @sizeOf(usize));
     stack_top.* = layout.THREAD_DESTROY;
 
-    return isr.Context.{
+    return isr.Context {
         .cs  = gdt.USER_CODE | gdt.USER_RPL,
         .ss  = gdt.USER_DATA | gdt.USER_RPL,
         .eip = entry_point,

--- a/kernel/thread.zig
+++ b/kernel/thread.zig
@@ -1,3 +1,4 @@
+// zig fmt: off
 const std = @import("std");
 const gdt = @import("gdt.zig");
 const isr = @import("isr.zig");
@@ -7,7 +8,7 @@ const vmem = @import("vmem.zig");
 const scheduler = @import("scheduler.zig");
 const x86 = @import("x86.zig");
 const Array = std.ArrayList;
-const List = std.IntrusiveLinkedList;
+const List = std.LinkedList;
 const Mailbox = @import("ipc.zig").Mailbox;
 const Message = std.os.zen.Message;
 const Process = @import("process.zig").Process;
@@ -19,15 +20,15 @@ const STACK_SIZE = x86.PAGE_SIZE;  // Size of thread stacks.
 var threads = Array(?*Thread).init(&mem.allocator);
 
 // List of threads inside a process.
-pub const ThreadList  = List(Thread, "process_link");
+pub const ThreadList  = List(void);
 // Queue of threads (for scheduler and mailboxes).
-pub const ThreadQueue = List(Thread, "queue_link");
+pub const ThreadQueue = List(void);
 
 // Structure representing a thread.
 pub const Thread = struct {
     // TODO: simplify once #679 is solved.
-    process_link: List(Thread, "process_link").Node,
-    queue_link:   List(Thread, "queue_link").Node,
+    process_link: List(void).Node,
+    queue_link:   List(void).Node,
 
     context: isr.Context,
     process: *Process,
@@ -62,8 +63,8 @@ pub const Thread = struct {
             .process      = process,
             .local_tid    = local_tid,
             .tid          = @intCast(u16, threads.len),
-            .process_link = ThreadList.Node.initIntrusive(),
-            .queue_link   = ThreadQueue.Node.initIntrusive(),
+            .process_link = ThreadList.Node.init({}),
+            .queue_link   = ThreadQueue.Node.init({}),
             .mailbox      = Mailbox.init(),
             .message_destination = undefined,
         };

--- a/kernel/tty.zig
+++ b/kernel/tty.zig
@@ -21,7 +21,7 @@ pub fn initialize() void {
 //     format: Format string.
 //     args: Parameters for format specifiers.
 //
-const Errors = error.{};
+const Errors = error {};
 pub fn print(comptime format: []const u8, args: ...) void {
     _ = fmt.format({}, Errors, printCallback, format, args);
 }

--- a/lib/tty.zig
+++ b/lib/tty.zig
@@ -15,7 +15,7 @@ pub const VGA_HEIGHT = 25;
 pub const VGA_SIZE   = VGA_WIDTH * VGA_HEIGHT;
 
 // Color codes.
-pub const Color = enum(u4).{
+pub const Color = enum(u4) {
     Black        = 0,
     Blue         = 1,
     Green        = 2,
@@ -35,7 +35,7 @@ pub const Color = enum(u4).{
 };
 
 // Character with attributes.
-pub const VGAEntry = packed struct.{
+pub const VGAEntry = packed struct {
     char:       u8,
     foreground: Color,
     background: Color,
@@ -58,7 +58,7 @@ pub fn disableCursor() void {
 }
 
 // VGA status.
-pub const VGA = struct.{
+pub const VGA = struct {
     vram:       []VGAEntry,
     cursor:     usize,
     foreground: Color,
@@ -74,7 +74,7 @@ pub const VGA = struct.{
     //     A structure holding the VGA status.
     //
     pub fn init(vram: usize) VGA {
-        return VGA.{
+        return VGA {
             .vram       = @intToPtr([*]VGAEntry, vram)[0..0x4000],
             .cursor     = 0,
             .foreground = Color.LightGrey,
@@ -198,7 +198,7 @@ pub const VGA = struct.{
     //     The requested VGAEntry.
     //
     fn entry(self: *VGA, char: u8) VGAEntry {
-        return VGAEntry.{
+        return VGAEntry {
             .char       = char,
             .foreground = self.foreground,
             .background = self.background,

--- a/servers/keyboard/main.zig
+++ b/servers/keyboard/main.zig
@@ -1,3 +1,4 @@
+// zig fmt: off
 const std = @import("std");
 const zen = std.os.zen;
 const Keyboard = zen.Server.Keyboard;
@@ -41,7 +42,7 @@ fn handleKeyEvent() void {
     if (waiting_thread) |thread| {
         // If a thread was blocked reading, send the character to it.
         waiting_thread = null;
-        zen.send(Message.to(thread, 0, char)
+        zen.send(&Message.to(thread, 0, char)
                         .as(Keyboard));
     } else {
         // Otherwise, save the character into the buffer.
@@ -53,15 +54,15 @@ fn handleKeyEvent() void {
 ////
 // Handle a read request from another thread.
 //
-fn handleRead(reader: *const MailboxId) void {
+fn handleRead(reader: MailboxId) void {
     if (buffer_start == buffer_end) {
         // If the buffer is empty, make the thread wait.
-        waiting_thread = reader.*;
+        waiting_thread = reader;
     } else {
         // Otherwise, fetch the first character from the buffer and send it.
         const char = buffer[buffer_start];
 
-        zen.send(Message.to(reader.*, 0, char)
+        zen.send(&Message.to(reader, 0, char)
                         .as(Keyboard));
 
         buffer_start = (buffer_start + 1) % buffer.len;
@@ -73,7 +74,7 @@ fn handleRead(reader: *const MailboxId) void {
 //
 pub fn main() void {
     // Instruct the kernel to send IRQ1 notifications to the Keyboard port.
-    zen.subscribeIRQ(1, Keyboard);
+    zen.subscribeIRQ(1, &Keyboard);
 
     // Receive messages from the Keyboard port.
     var message = Message.from(Keyboard);

--- a/servers/shell/main.zig
+++ b/servers/shell/main.zig
@@ -11,7 +11,7 @@ const warn = std.debug.warn;
 //
 pub fn main() void {
     var stdin_file = io.getStdIn() catch unreachable;
-    var stdin = &io.FileInStream.init(&stdin_file).stream;
+    var stdin = &stdin_file.inStream().stream;
     var buffer: [1024]u8 = undefined;
 
     warn("\n");
@@ -52,7 +52,6 @@ fn execute(command: []u8) void {
 //
 fn readLine(stream: var, buffer: []u8) usize {
     // TODO: change the type of stream when #764 is fixed.
-
     var i: usize = 0;
     var char: u8 = 0;
 
@@ -73,24 +72,23 @@ fn readLine(stream: var, buffer: []u8) usize {
         }
     }
 
-    return i - 1;  // Exclude \n.
+    return i - 1; // Exclude \n.
 }
-
 
 //////////////////////////
 ////  Shell commands  ////
 //////////////////////////
 
 fn clear() void {
-    zen.send(Message.to(Server.Terminal, 0));
+    zen.send(&Message.to(Server.Terminal, 0));
 }
 
 fn help() void {
     warn("{}\n\n",
-         \\List of supported commands:
-         \\    clear      Clear the screen
-         \\    help       Show help message
-         \\    version    Show Zen version
+        \\List of supported commands:
+        \\    clear      Clear the screen
+        \\    help       Show help message
+        \\    version    Show Zen version
     );
 }
 

